frontend/src/metabase/home/homepage/actions.ts

```diff
@@ -3,23 +3,23 @@ import { updateSetting } from "metabase/admin/settings/settings";
 
 export const HIDE_DATA = "metabase/home/homepage/HIDE_DATA";
 export const hideData = createThunkAction(HIDE_DATA, function() {
-  return async function(dispatch) {
+  return async function(dispatch: any) {
     const setting = { key: "show-homepage-data", value: false };
     await dispatch(updateSetting(setting));
   };
 });
 
 export const HIDE_XRAYS = "metabase/home/homepage/HIDE_XRAYS";
 export const hideXrays = createThunkAction(HIDE_XRAYS, function() {
-  return async function(dispatch) {
+  return async function(dispatch: any) {
```

Comment:
Not sure if it applies here but you can type a `key/value` object response using (useful for JSON for instance):
```
{
  [key: string]: any | orAnyOtherType;
}
```